### PR TITLE
Bugfix/restart split domain

### PR DESCRIPTION
**Pull request by arbennett**
This fixes a bug in using restart files if you use the `-g startGRU countGRU` flag. Previously the initial conditions were always selected starting at the first index instead of at `startGRU`. Note that this does break the ability to use subsetted restart files.

### CODE REVIEW

File: build/source/netcdf/read_icond.f90 [hunk 1]
```diff
@@ -292,10 +294,9 @@ subroutine read_icond(iconFile,                      & ! intent(in):    name of
     ! get the index in the file: single HRU
     if(restartFileType==singleHRU)then
      ixFile = 1  ! use for single HRU restart file
-
     ! get the index in the file: multi HRU
     else
-     ixFile = iHRU_local
+     ixFile = startGRU + iHRU_local - 1
```
Comment (bartnijssen) [inline build/source/netcdf/read_icond.f90:299]: Why the `-1` offset relative to `startGRU`? That seems a C things (if `startGRU` starts at ` and indices start at 0). Is this correct? I guess if both `startGRU` and `iHRU_local` are 1 this works.

**Review by bartnijssen — COMMENTED** [review]
@arbennett : Just see the question and provide a response.

File: build/source/netcdf/read_icond.f90 (hunk 1 above)
Comment (andywood) [reply build/source/netcdf/read_icond.f90:299]: right.  I think this is the equivalent of using ixFile = iHRU_global, incidentally.  using iHRU_local was the bug before.

**Review by andywood — APPROVED** [review]

File: build/source/netcdf/read_icond.f90 (hunk 1 above)
Comment (arbennett) [reply build/source/netcdf/read_icond.f90:299]: Exactly, basically it's just accounting for 2 things that both start at 1

**Pull request merged**